### PR TITLE
chgrp/chown: move common code to `uucore`

### LINKED ISSUE #2565: `chgrp`/`chown`: reduce code duplication
`chgrp` and `chown` do mostly the same - and in fact their implementation is also mostly the same. It looks like one implementation was copy-pasted from the other one with very slight modifications...
I think we can do a lot better here and share more code between `chgrp` and `chown`, which would improve maintainability.

### PR DESCRIPTION
**Pull request by miDeb**
chgrp does mostly the same as chown.
By making chown a bit more configurable we can reuse its code for chgrp.
Closes #2565.
Depends on #2471 for correct error messages.

### CODE REVIEW

File: src/uu/chgrp/Cargo.toml [hunk 1]
```diff
@@ -18,7 +18,7 @@ path = "src/chgrp.rs"
 clap = { version = "2.33", features = ["wrap_help"] }
 uucore = { version=">=0.0.9", package="uucore", path="../../uucore", features=["entries", "fs", "perms"] }
 uucore_procs = { version=">=0.0.6", package="uucore_procs", path="../../uucore_procs" }
-walkdir = "2.2"
+uu_chown = { version=">=0.0.6", package="uu_chown", path="../chown"}
```
Comment (jhscheer) [inline src/uu/chgrp/Cargo.toml:21]: I guess it's just a matter of taste, but I would move duplicate code to `uucore`.

Comment (miDeb) [reply src/uu/chgrp/Cargo.toml:21]: Makes sense, I'll do that instead

**Comment by sylvestre** [comment]
terrific idea, bravo!
I guess you noticed the Mac errors, right ?

**Comment by miDeb** [comment]
As far as I can tell those test failures are becuase the error messages now start with `uucore: ` instead of `chown: ` due to the refactoring. This should be fixed by #2471

**Comment by sylvestre** [comment]
Could you please rebase this patch now that the other has been merged?

**Pull request merged**